onnxruntime/core/optimizer/utils.h

```diff
@@ -4,15 +4,26 @@
 #pragma once
 
 #include "core/graph/onnx_protobuf.h"
+#include "core/graph/graph.h"
 
 namespace onnxruntime {
+class Graph;
+class NodeArg;
+
 namespace optimizer_utils {
 
 // Check if TensorProto contains a floating point type.
-static bool IsFloatingPointDataType(const ONNX_NAMESPACE::TensorProto& tensor_proto) {
-  return !(tensor_proto.data_type() != ONNX_NAMESPACE::TensorProto_DataType_FLOAT &&
-           tensor_proto.data_type() != ONNX_NAMESPACE::TensorProto_DataType_FLOAT16 &&
-           tensor_proto.data_type() != ONNX_NAMESPACE::TensorProto_DataType_DOUBLE);
-}
+bool IsFloatingPointDataType(const ONNX_NAMESPACE::TensorProto& tensor_proto);
+
+/** Check a input is constant with expected float value.
+@remarks only support float16, float and double scalar.
+*/
+bool CheckConstantInput(const onnxruntime::Graph& graph, const onnxruntime::NodeArg& input_arg, float expected_value);
+
```

Comment:
Name doesn't feel descriptive enough. You're checking a node input is constant and has a specific value.